addons/mail/static/src/core/common/message.js

```diff
@@ -243,7 +243,15 @@ export class Message extends Component {
         }
 
         if (this.message.author) {
-            return this.message.author.avatarUrl;
+            const params = {};
+            if (this.props.thread) {
+                Object.assign(params, {
+                    thread_model: this.props.thread.model,
+                    thread_id: this.props.thread.id,
+                    ...this.props.thread.rpcParams,
+                });
+            }
+            return this.message.author.avatarUrl(params);
```

Comment:
This look much simpler to me, you don't need the `const params` and the `Object.assign`

```suggestion
            if (this.props.thread) {
                return this.message.author.avatarUrl({
                    thread_model: this.props.thread.model,
                    thread_id: this.props.thread.id,
                    ...this.props.thread.rpcParams,
                });
            }
            return this.message.author.avatarUrl();
```